### PR TITLE
fix(settings): disable LDAP by default

### LINKED ISSUE #1189: warning message appearing in logs when logging in
Seeing this during the e2e runs, likely introduced after #1156:

> WARNING:django_auth_ldap:Caught LDAPError while authenticating test-269851327: SERVER_DOWN({'desc': "Can't contact LDAP server"},)
> 10.88.2.87 "POST /v2/auth/login/ HTTP/1.1" 200 52 "Deis Client v2.9.1-3e324ea"

The login request is accepted but it looks like the controller is attempting to communicate with an LDAP server that does not exist. This functionality should be shut off by default.

### PR DESCRIPTION
**Pull request by bacongobbler**
fixes #1189

### CODE REVIEW

**Comment by bacongobbler** [comment]
Before and after:

```
$ ./manage.py test --settings=api.settings.testing --noinput api.tests.test_auth.AuthTest.test_auth_no_ldap_by_default
Creating test database for alias 'default'...
F
======================================================================
FAIL: test_auth_no_ldap_by_default (api.tests.test_auth.AuthTest)
Ensure that LDAP authentication is disabled by default.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.5/lib/python3.5/unittest/mock.py", line 1157, in patched
    return func(*args, **keywargs)
  File "/home/bacongobbler/go/src/github.com/deis/controller/rootfs/api/tests/test_auth.py", line 384, in test_auth_no_ldap_by_default
    mock_logger.warning.assert_not_called()
  File "/usr/lib/python3.5/lib/python3.5/unittest/mock.py", line 773, in assert_not_called
    raise AssertionError(msg)
AssertionError: Expected 'warning' to not have been called. Called 1 times.

----------------------------------------------------------------------
Ran 1 test in 0.106s

FAILED (failures=1)
Destroying test database for alias 'default'...

$ ./manage.py test --settings=api.settings.testing --noinput api.tests.test_auth.AuthTest.test_auth_no_ldap_by_default
Creating test database for alias 'default'...
.
----------------------------------------------------------------------
Ran 1 test in 0.098s

OK
Destroying test database for alias 'default'...
```

File: rootfs/api/settings/production.py [hunk 1]
```diff
@@ -367,31 +366,33 @@
...
-AUTH_LDAP_GLOBAL_OPTIONS = {
-    ldap.OPT_X_TLS_REQUIRE_CERT: False,
-    ldap.OPT_REFERRALS: False
-}
-AUTH_LDAP_ALWAYS_UPDATE_USER = True
-AUTH_LDAP_MIRROR_GROUPS = True
-AUTH_LDAP_FIND_GROUP_PERMS = True
-AUTH_LDAP_CACHE_GROUPS = False
+if LDAP_ENDPOINT:
+    AUTHENTICATION_BACKENDS += "django_auth_ldap.backend.LDAPBackend"
```
Comment (mboersma) [inline rootfs/api/settings/production.py]: I think this needs to be changed to add the tuples together:
```python
>>> AUTHENTICATION_BACKENDS
('django.contrib.auth.backends.ModelBackend', 'guardian.backends.ObjectPermissionBackend')
>>> AUTHENTICATION_BACKENDS += "django_auth_ldap.backend.LDAPBackend"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate tuple (not "str") to tuple
>>> AUTHENTICATION_BACKENDS = ("django_auth_ldap.backend.LDAPBackend",) + AUTHENTICATION_BACKENDS
>>> AUTHENTICATION_BACKENDS
('django_auth_ldap.backend.LDAPBackend', 'django.contrib.auth.backends.ModelBackend', 'guardian.backends.ObjectPermissionBackend')
```

Comment (bacongobbler) [inline rootfs/api/settings/production.py]: good catch.

Comment (bacongobbler) [inline rootfs/api/settings/production.py]: fixed

**Pull request merged**